catshadow/client.go

```diff
@@ -1441,16 +1447,16 @@ func (c *Client) getSpoolWriteDescriptor() *memspoolclient.SpoolWriteDescriptor
 func (c *Client) goOffline() error {
 	c.connMutex.Lock()
 	defer c.connMutex.Unlock()
-	if c.connecting {
+	if c.connected == StateConnecting {
 		return errors.New("Offline() does not cancel Online()")
 	}
 
-	if !c.online || c.session == nil {
```

Comment:
what happened to the `c.session` check, is it an error if we are `StateConnecting` and have a `nil` session? or `StateOffline` but a non-`nil` `c.session`?

Reply:
I think it should never be in StateConnecting with a non-nil session

Reply:
the check is removed since doOnline does the same check, and there is only one place where session is set or nil'd while the lock is held

Reply:
Hmm I can't find `doOnline` anywhere, which file is that in?

Reply:
Sorry it's not: because `if c.connected == StateConnecting { return errors `

Reply:
Is this resolved or can you propose a change to be made?